src/robusta/core/model/events.py

```diff
@@ -28,7 +29,7 @@ class ExecutionEventBaseParams(BaseModel):
 # (note that we need to integrate with dataclasses because of hikaru)
 @dataclass
 class ExecutionBaseEvent:
-    findings: List[Finding] = field(default_factory=lambda: [])
+    sink_findings: Dict[str, List[Finding]] = field(default_factory=lambda: defaultdict(list))
```

Comment:
Looks good, but I would update the variable names here and add some documentation about non-obvious stuff. I think the current behaviour is:

`sink_findings` - this is something like `shared_findings` - it includes all Findings generated so far on the current trigger-event and it is shared between all ExecutionBaseEvents of different types. It can contain findings for sinks that the current ExecutionBaseEvent doesn't care about.

`named_sinks` - this is something like `my_sinks` - it contains a subset of `sink_findings.keys()` which the current ExecutionBaseEvent actually cares about and is configured for.

Just to emphasize, the behaviour here looks good. I would just document more.

Reply:
I think the name `sink_findings` is good, because it's the findings collected to be sent to each sink.
I'll add some more docs about the behavior which isn't self explanatory